features/grpc/exporter/src/test/java/org/opennms/features/grpc/exporter/SpogInventoryServiceSyncIT.java

```diff
@@ -0,0 +1,570 @@
+
+        // Step 1: Wait for the channel to be ready and set up the initial inventory update
+        waitForChannelToBeReady();
+
+        spogInventoryService = new SpogInventoryService(nodeDao, runtimeInfo, client, new MockSessionUtils(), 0, true);
+        inventoryExporter = new InventoryExporter(eventSubscriptionService, nodeDao, spogInventoryService);
+
+        final var nodeEvent = MockEventUtil.createNodeUpEventBuilder("test", databasePopulator.getNode2()).getEvent();
+        eventdIpcMgr.sendNow(nodeEvent);
+        this.nodeId = nodeEvent.getNodeid();
+
+        AtomicReference<Boolean> inventoryFlag = new AtomicReference<>(false);
+        await().pollInterval(2000, TimeUnit.MILLISECONDS)
+                .atMost(1, TimeUnit.MINUTES)
+                .until(() -> {
+                    try {
+                        inventoryExporter.onEvent(ImmutableMapper.fromMutableEvent(nodeEvent));
+                        inventoryFlag.set(serverResponseFuture.get(10, TimeUnit.SECONDS));
+                        return inventoryFlag.get();
+                    } catch (Exception e) {
+                        LOG.error(e.getMessage());
+                        return false;
+                    }
+                });
+        assertTrue("The initial response was not completed successfully.", inventoryFlag.get());
+
+        // Step 2: Stop the server and client to simulate a restart
+        if (server != null) {
+            client.stop();
+            server.shutdown();
+            // Wait for the server to fully shutdown
+            assertTrue("Server failed to shut down properly", server.awaitTermination(30, TimeUnit.SECONDS));
+        }
+
+        // Step 3: Restart the server and client
+        initializeAndStartServer();
+        initializeAndStartClient();
+        // Step 4: await channel to be ready
+        waitForChannelToBeReady();
+        // Step 5: reset and await
+        inventoryFlag.set(false);
+        await().pollInterval(2000, TimeUnit.MILLISECONDS)
+                .atMost(1, TimeUnit.MINUTES)
+                .until(() -> {
+                    try {
+                        inventoryExporter.onEvent(ImmutableMapper.fromMutableEvent(nodeEvent));
+                        inventoryFlag.set(serverResponseFuture.get(10, TimeUnit.SECONDS));
+                        return inventoryFlag.get();
+                    } catch (Exception e) {
+                        LOG.error(e.getMessage());
+                        return false;
+                    }
+                });
+        // Step 6: Assert that the client was able to reconnect and send the update after server restart
+        assertTrue("The response after server recovery was not completed successfully.", inventoryFlag.get());
+        System.out.println("Client successfully recovered and sent inventory update after server restart.");
+    }
+
+    @Test
+    public void testInventoryDataToBeSent() {
```

Comment:
One test case should validate whether all Inventory data is sent to server for the first time even without receiving any event.